pkg/cmd/compose/compose.go

```diff
@@ -35,14 +35,29 @@ import (
 	"github.com/containerd/nerdctl/v2/pkg/composer/serviceparser"
 	"github.com/containerd/nerdctl/v2/pkg/imgutil"
 	"github.com/containerd/nerdctl/v2/pkg/ipfs"
+	"github.com/containerd/nerdctl/v2/pkg/lockutil"
 	"github.com/containerd/nerdctl/v2/pkg/netutil"
 	"github.com/containerd/nerdctl/v2/pkg/referenceutil"
 	"github.com/containerd/nerdctl/v2/pkg/signutil"
 	"github.com/containerd/nerdctl/v2/pkg/strutil"
 )
 
+//nolint:unused
+var locked *os.File
+
 // New returns a new *composer.Composer.
 func New(client *containerd.Client, globalOptions types.GlobalCommandOptions, options composer.Options, stdout, stderr io.Writer) (*composer.Composer, error) {
+	// Compose right now cannot be made safe to use concurrently, as we shell out to nerdctl for multiple operations,
+	// preventing us from using the lock mechanisms from the API.
+	// This here imposes a global lock, effectively preventing multiple compose commands from being run in parallel and
+	// preventing some of the problems with concurrent execution.
+	// This should be removed once we have better, in-depth solutions to make this concurrency safe.
+	var err error
+	locked, err = lockutil.Lock(globalOptions.DataRoot)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
Where is unlock?

Reply:
When `nerdctl compose` exits, it will release the lock - we do not release explicitly.

The reason I did no do it explicitly is because (I think) we would have to have that code inside *all* compose commands, which would make for a large/more intrusive patch.
This makes here for a much simpler patch.

Reply:
Could you add code comments to explain that

Reply:
Also wondering if we can just add *Composer.Close

Reply:
> Also wondering if we can just add `*Composer.Close()`.
> The caller of `compose.New` should have `defer c.Close()`

Sure.

Is it worth the effort and extra code though?